### DISCUSSION TITLE
How to add custom method?

### DISCUSSION THREAD

**Discussion by BoThe1K** [WalletConnect/walletconnect-monorepo#4326] [Developer Support]
Hey guys! I want to add custom method which will be triggered and forwarded to the wallet (like `eth_sendTransaction`, `personal_sign` etc., so method like `xxx_signData` which will go directly to the wallet with session request and not will trigger `web3` rpc. Is there any way to make this work without overriding of internals? (I need it mostly for our wallet to be connected with wallet connect and use some our apis)

Using `@web3modal/ethers5/react` during integration.

**Comment by glitch-txs** [comment]
This is not possible in Web3Modal, would you can do it with some of these providers: https://docs.walletconnect.com/advanced/providers/ethereum
